### PR TITLE
libceed: remove custom phases, slight refactor of variants

### PR DESCRIPTION
**Pull request by alalazo**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libceed/package.py
Comment (jedbrown) [inline var/spack/repos/builtin/packages/libceed/package.py:141]: I do we assume people will have `prove` (from perl) or should we use `make test` here?

**Review by jedbrown — APPROVED** [review]
Looks fine to me, thanks. @jeremylt Should we bundle this into #29710 ?

**Comment by alalazo** [comment]
@jedbrown Fine if I merge this when CI succeeds? Or should we wait for some other review too?

File: var/spack/repos/builtin/packages/libceed/package.py
Comment (jeremylt) [inline var/spack/repos/builtin/packages/libceed/package.py:39]: Perhaps it would be easier to combine the two PRs? Technically the OCCA backend no longer works, as of libCEED v0.9.0, so libceed@develop + occa@develop is broken and should not be available to users

Comment (alalazo) [reply var/spack/repos/builtin/packages/libceed/package.py:39]: @jeremylt Let me know what you'd prefer. I can remove that line, or leave it and review one of the PRs you linked that does that. My point here was really getting rid of the `phases` attribute (I'm trying to check how many of the ~80 packages that declare custom phases really need them).

Comment (jeremylt) [reply var/spack/repos/builtin/packages/libceed/package.py:39]: @jedbrown, how about we merge this and I'll update my MR accordingly?

**Review by jeremylt — APPROVED** [review]

File: var/spack/repos/builtin/packages/libceed/package.py
Comment (jedbrown) [reply var/spack/repos/builtin/packages/libceed/package.py:39]: Sounds good.

**Comment by alalazo** [comment]
Thanks! Let me know if I can help reviewing/merging related PRs

**Comment by v-dobrev** [comment]
This change actually broke the install: in the `install` phase, the `install` target is not given to `make`. This is what's in the log file:
```console
==> libceed: Executing phase: 'install'
==> [2022-04-13-19:10:12.506611] 'make' '-j8' 'prefix=/Users/dobrev1/spack/opt/spack/darwin-bigsur-skylake/apple-clang-13.0.0/libceed-0.8-tut322kuz2nrtfqhio5ofkfhl2pvg6pf' 'V=1' 'OPT=-O3 -g -march=native -ffp-contract=fast -fopenmp-simd' 'CUDA_DIR=/disable-cuda'
make: 'lib' with optional backends: /cpu/self/avx/serial /cpu/self/avx/blocked
make: Nothing to be done for `lib'.
```
I can probably fix this for `libceed` (e.g. in #29710) but it may affect other packages too.

**Comment by v-dobrev** [comment]
I suspect the intent of `install_targets` is to override the default `install` target and/or add additional command line args, so maybe this just affects `libceed` and we simply need to explicitly add the `install` target?

**Comment by v-dobrev** [comment]
I pushed a fix for `libceed` in #29710.

**Comment by alalazo** [comment]
@v-dobrev You're right. It seems I missed the target during the refactor. How long before #29710 is ready for review? We plan to release Spack v0.18.0 in 4 weeks at most, so in case we can have your fix merged quickly as a separate PR.

**Comment by v-dobrev** [comment]
@alalazo, I think that #29710 will be ready for review in a week or two, so it should be done before the v0.18.0 release.

**Pull request merged**
